R/get_pacfin_expansions.R

```diff
@@ -0,0 +1,211 @@
+#'   in Stock Synthesis, but expansions are done in pounds because fish weights
+#'   are in pounds. Thus, catches also need to be in pounds and will be
+#'   converted as such.
+#' @param stratification.cols A vector of column names in `Pdata` that you want
+#'   to use as strata. These will match the way in which the catches are
+#'   transformed from long to wide prior to inputting them into this function.
+#'   If you leave this argument empty, then `Pdata` must already have a column
+#'   named `stratification`. The function will look in the column names of the
+#'   `Catch` data to determine the appropriate separator to use between columns
+#'   when pasting the words together, which is done using [apply] and [paste].
+#'   Historically, it was mandatory to make this column yourself, but in 2021,
+#'   this input argument was added to reduce the number of extraneous calls that
+#'   were needed between functions. You can use as many levels of stratification
+#'   as you want except year because it is already included in the call to
+#'   [stats::aggregate].
+#' @template savedir
+#'
+#' @examples
+#' \dontrun{
+#' # Calculate the weight--length parameters for input to this function
+#' bds_survey <- nwfscSurvey::pull_bio(
+#'   common_name = "widow rockfish",
+#'   survey = "NWFSC.Combo"
+#'   )
+#' pars <- nwfscSurvey::estimate_weight_length(
+#'   data = bds_survey,
+#'   col_length = "length_cm",
+#'   col_weight = "weight_kg",
+#'   verbose = FALSE
+#'  )
+#'  
+#' expanded_comps <- get_pacfin_expansions(
+#'   Pdata = bds_cleaned,
+#'   Catch = catch_dataframe,
+#'   weight_length_estimates = pars,
+#'   Units = "MT",
+#'   Comps = "LEN",
+#'   maxExp = 0.95)
+#' }
+#' @return
+#' A `data.frame` with expanded data up to the trip and total catch level.
+#'
+#'
+get_pacfin_expansions <- function(
+    Pdata,
+    Catch,
+    weight_length_estimates,
+    stratification.cols,
+    Units = "MT",
+    maxExp = 0.95,
+    Exp_WA = TRUE,
+    verbose = TRUE,
+    savedir = NULL) {
+  nwfscSurvey::check_dir(dir = savedir, verbose = verbose)
+  
+  # Check weight-length data frame
+  required_cols <- c("sex", "A", "B")
+  if (any(!required_cols %in% colnames(weight_length_estimates))){
+    cli::cli_abort(
+      "The weight_length_estimates dataframe is required to have three columns 
+    named {required_cols}")
+  }
+  required_sexes <- c("female", "male", "all") 
+  if (any(!required_sexes %in% weight_length_estimates[,"sex"])){
+    cli::cli_abort(
+      "The weight_length_estimates dataframe is required to have three rows in  
+    the sex column for {required_sexes}")
+  }
+  if (verbose) {
+    cli::cli_inform(
+      "The units for the Catch dataframe is specified to be in {Units}"
+    )
+    if (Units == "MT") {
+      cli::cli_inform(
+        "Catches will be converted to pounds for the second-stage expansion."
+      )
+    }
+  }
+  fa = weight_length_estimates |>
```

Comment:
I see a few places where this function doesn't match the standard style (e.g. `=` vs `<-`), but the github action will take care of that.

Reply:
Good catch.  I will fix this.